src/superstring/core.clj

```diff
@@ -29,52 +32,52 @@
   "Create an alias for all public vars in ns in this ns."
   [namespace]
   `(do ~@(map
-          (fn [n] `(defalias ~(.sym n) ~(symbol (str (.ns n)) (str (.sym n)))))
+          (fn [^clojure.lang.Var n] `(defalias ~(.sym n) ~(symbol (str (.ns n)) (str (.sym n)))))
           (vals (ns-publics namespace)))))
 
 (alias-ns clojure.string)
-(def substring clojure.core/subs)
+(def ^{:tag String} substring clojure.core/subs)
 
-(defn ^Long length
+(defn length
   "Return the length of s."
-  [^String s]
-  {:pre [(string? s)]
-   :post [(integer? %)]}
-  (.length s))
+  (^long [^String s]
+    {:pre [(string? s)]
+     :post [(integer? %)]}
+    (.length s)))
 
-(defn ^Long index-of
+(defn index-of
   "Return the starting position of the first occurrence of needle in s or nil.
 
   If start is provided, start the search at that position in s."
-  ([^String s needle]
+  (^Long [^String s needle]
    {:pre [(string? s) (string? needle)]
     :post [(or (integer? %) (nil? %))]}
    (index-of s needle 0))
-  ([^String s ^String needle start]
+  (^Long [^String s ^String needle ^long start]
```

Comment:
You need to use `Long` if you want to use nil as a "not found" value. You could use `long` if you want to switch to -1 as a return value, but that would be a breaking change so I left it as-is at the moment.